protocol/message.go

```diff
@@ -120,3 +130,70 @@ func NewMonitoringProof(ap []*m.AuthenticationPath,
 		},
 	}, e
 }
+
+// Verify implementations
+
+func (df *DirectoryProof) Verify(uname string, key []byte,
+	savedSTR []byte, signKey sign.PublicKey) ErrorCode {
+	ap := df.AP
+	str := df.STR
+	tb := df.TB
+
+	if e := verifySTR(signKey, str, savedSTR); e != Passed {
+		return e
+	}
+
+	// verify TB's Signature
+	if tb != nil {
```

Comment:
The client probably needs to know beforehand whether it expects `tb` to be `nil` or not.